ebos/eclgenericoutputblackoilmodule.cc

```diff
@@ -139,6 +139,39 @@ EclGenericOutputBlackoilModule(const EclipseState& eclState,
         std::string key_pattern = "R" + EclString(phase) + "*";
         this->regionNodes_[phase] = summaryConfig_.keywords(key_pattern);
     }
+
+    // Check if FLOWS is set in any RPTRST in the schedule
+    const int totalSteps = schedule_.size();
+    enableFlows_ = false;
+    enableFlowsn_ = false;
+    anyFlows_ = false;
+    for (int step = 0; step < totalSteps; step++) {
+        if(anyFlows_ == true)
+            continue;
+
+        std::map<std::string, int> rstKeywords = schedule_.rst_keywords(step);
+        for (auto& [keyword, should_write] : rstKeywords) {
+            if (keyword == "FLOWS") {
+                anyFlows_ = true;
+            }
+        }
+    }
+
+    // Check if FLORES is set in any RPTRST in the schedule
+    anyFlores_ = false;
+    enableFlores_ = false;
+    enableFloresn_ = false;
+    for (int step = 0; step < totalSteps; step++) {
+        if(anyFlores_ == true)
+            continue;
+
+        std::map<std::string, int> rstKeywords = schedule_.rst_keywords(step);
+        for (auto& [keyword, should_write] : rstKeywords) {
+            if (keyword == "FLORES") {
+                anyFlores_ = true;
+            }
+        }
+    }
```

Comment:
You can probably simplify this section further, and avoid copying the [`RSTConfig::keywords`](https://github.com/OPM/opm-common/blob/b83f10d2d70d160e50a6df209eeb8f7cffd4c774/opm/input/eclipse/Schedule/RSTConfig.hpp#L217) map, if you do something along the lines of
```C++
for (const auto& block : this->schedule_) { // Uses Schedule::begin() and Schedule::end()
    const auto& rstkw = block.rst_config().keywords;

    if (! anyFlows_) {
        anyFlows_ = rstkw.find("FLOWS") != rstkw.end();
    }

    if (! anyFlores_) {
        anyFlores_ = rstkw.find("FLORES") != rstkw.end();
    }

    if (anyFlows_ && anyFlores_) {
        // Terminate report step loop early if both FLOWS and FLORES
        // have been set at some point as there's no need to search
        // any further in that case.
        break;
    }
}
```